aip/0123.md

```diff
@@ -0,0 +1,68 @@
+---
+aip:
+  id: 123
+  state: reviewing
+  created: 2019-05-12
+permalink: /123
+redirect_from:
+  - /0123
+---
+
+# Resource types
+
+Most APIs expose _resources_ (their primary nouns) which users are able to
+create, retrieve, and manipulate. APIs are allowed to name their resource types
+as they see fit, and are only required to ensure uniqueness within that API.
+This means that it is possible (and often desirable) for different APIs to use
+the same type name. For example, a Memcache and Redis API would both want to
+use `Instance` as a type name.
+
+When mapping the relationships between APIs and their resources, however, it
+becomes important to have a single, globally-unique type name. Additionally,
+tools such as Kubernetes or GraphQL interact with APIs from multiple providers.
+
+## Terminology
+
+In the guidance below, we use the following terms:
+
+- **Service Name:** This is the name defined in the [service configuration][].
+  This usually (but not necessarily) matches the hostname that users use to
+  call the service. Example: `pubsub.googleapis.com`. This is equivalent to an
+  [API Group][] in Kubernetes.
+- **Type:** This is the name used for the type within the API. This usually
+  (but not necessarily) matches the name of the protobuf message. This is
+  equivalent to an [Object][] in Kubernetes.
+
+## Guidance
+
+APIs **must** define a resource type for each resource in the API, according to
+the following pattern: `{Service Name}/{Type}`. The type name **must** be
+singular and use PascalCase (UpperCamelCase).
+
+### Examples
+
+Examples of resource types include:
+
+- `pubsub.googleapis.com/Topic`
+- `pubsub.googleapis.com/Subscription`
```

Comment:
Possibly add an example with two words?

Reply:
Discussed with @jskeet in person. Neither of us could think of one that we were confident in. We will add this later. :-)